Sources/PackageCollectionGenerator/PackageMetadataProviders/GitLabPackageManagerMetadataProvider.swift

```diff
@@ -0,0 +1,170 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Package Collection Generator open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift Package Collection Generator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Package Collection Generator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Dispatch
+import Foundation
+
+import TSCBasic
+import Basics
+import PackageCollectionsModel
+import Utilities
+
+struct GitLabPackageMetadataProvider: PackageMetadataProvider {
```

Comment:
Please add `GitLabPackageMetadataProviderTests` (see `GitHubPackageMetadataProviderTests` for example) that tests with real sample response payloads.

Reply:
Yes, I will add them.